mongo/options/autoencryptionoptions.go

```diff
@@ -91,6 +98,43 @@ func (a *AutoEncryptionOptions) SetExtraOptions(extraOpts map[string]interface{}
 	return a
 }
 
+// SetTLSConfig applies custom TLS options to configure connections with each KMS provider.
+func (a *AutoEncryptionOptions) SetTLSConfig(tlsOptsMap map[string]map[string]interface{}) (*AutoEncryptionOptions, error) {
+	a.TLSConfig = make(map[string]tls.Config)
+
+	for provider, tlsOpts := range tlsOptsMap {
+		var cfg tls.Config
+		
+		for cert := range tlsOpts {
+			var err error
+			switch cert {
+			case "tlsCertificateKeyFile":
```

Comment:
Unsure if I should define a `Type` for the TLS option field names.

Reply:
I'm not sure if a type + constants would make this API clearer. I think we should at least provide a detailed comment describing what values to use, similar to the [ClientOptions.SetTLSConfig docs](https://github.com/mongodb/mongo-go-driver/blob/48ca24629ba874b6b64c3d460c446f71b72f9c2b/mongo/options/clientoptions.go#L720) or [pymongo's AutoEncryptionOpts docs](https://pymongo.readthedocs.io/en/stable/api/pymongo/encryption_options.html).